cogboard-app/src/main/kotlin/com/cognifide/cogboard/ssh/SSHClient.kt

```diff
@@ -0,0 +1,85 @@
+package com.cognifide.cogboard.ssh
+
+import com.cognifide.cogboard.CogboardConstants
+import com.cognifide.cogboard.ssh.auth.SSHAuthData
+import com.cognifide.cogboard.ssh.session.SessionStrategyFactory
+import com.jcraft.jsch.ChannelExec
+import com.jcraft.jsch.JSch
+import com.jcraft.jsch.JSchException
+import com.jcraft.jsch.Session
+import io.vertx.core.buffer.Buffer
+import io.vertx.core.json.JsonObject
+import io.vertx.core.logging.Logger
+import io.vertx.core.logging.LoggerFactory
+import java.io.InputStream
+import java.nio.charset.Charset
+
+class SSHClient(private val config: JsonObject) {
+    private var session: Session? = null
+    private var jsch: JSch? = null
+
+    private fun openSession() {
+        val authData = SSHAuthData(config)
+        val jsch = JSch()
+        val session = SessionStrategyFactory(jsch).create(authData).initSession()
```

Comment:
Does this code throw any known exceptions? Maybe it should be handled here somehow.